doc/source/ray-air/examples/pytorch_resnet_batch_prediction.ipynb

```diff
@@ -0,0 +1,398 @@
+      "Read progress: 100%|██████████| 50/50 [00:00<00:00, 3996.02it/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "#@title +\n",
+    "if SMOKE_TEST:\n",
+    "    ds = ds.limit(1000)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that the label for images are their corresponding directory names (e.g. n01728920). To find the indices of our model output that correspond to these names, we'll need to download a mapping from the s3 bucket (`imagenet_class_index.json`). The actual class names can also be found from this file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import boto3\n",
+    "from botocore import UNSIGNED\n",
+    "from botocore.client import Config\n",
+    "\n",
+    "# Download mapping file from S3\n",
+    "s3 = boto3.client(\"s3\", config=Config(signature_version=UNSIGNED))\n",
+    "s3.download_file(\n",
+    "    \"air-example-data-2\",\n",
+    "    \"imagenette2/imagenet_class_index.json\",\n",
+    "    \"/tmp/imagenet_class_index.json\",\n",
+    ")\n",
+    "\n",
+    "# Build mappings\n",
+    "idx_to_class = json.load(open(\"/tmp/imagenet_class_index.json\", \"r\"))\n",
+    "class_to_idx = {cls_name: int(index) for index, (cls_name, _) in idx_to_class.items()}\n",
+    "idx_to_class_name = {int(index): text for index, (_, text) in idx_to_class.items()}\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, let's define a preprocessor to crop and normalize images, as well as convert the class names to indices with the map that we just constructed. We'll use a {class}`BatchMapper <ray.data.preprocessors.BatchMapper>` to implement this custom preprocessing logic. `BatchMapper` takes in a user defined function and applies that as the data transformation to the input dataset, before feeding it into the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from torchvision import transforms\n",
+    "from ray.data.preprocessors import BatchMapper\n",
+    "from typing import Dict\n",
+    "\n",
+    "\n",
+    "def preprocess(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:\n",
+    "    transform = transforms.Compose(\n",
+    "        [\n",
+    "            transforms.ToPILImage(),\n",
+    "            transforms.Resize(256),\n",
+    "            transforms.CenterCrop(224),\n",
+    "            transforms.ToTensor(),\n",
+    "            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),\n",
+    "        ]\n",
+    "    )\n",
+    "    batch[\"image\"] = np.array([transform(img).numpy() for img in batch[\"image\"]])\n",
+    "    batch[\"label\"] = np.array([class_to_idx[cls_name] for cls_name in batch[\"class\"]])\n",
```

Comment:
Actually this label mapping contains 1000 k-v pairs, which will be used for evaluating model predictions later. Not sure if it will be faster to iterate through the dictionary. I can try with `np.vectorize()` and this should be a more elegant way.

Reply:
can we reduce the size of the dict to just the 10 classes that are being used? that's probably cleaner than np.vectorize

Reply:
hmm. We can actually create a separate label mapping, but I need to create another mapping file and upload to S3.
And also we still need the 1000-class mapping in the evaluation section.

Reply:
```python
batch["label"] = np.vectorize(class_to_idx.__getitem__)(batch["class"])
```

Current solution is like this. It's not that messy I guess?

Reply:
ok this is good!

Reply:
ok this is good! let's make sure to add a comment explaining what this is doing

Reply:
Added one comment for this mapping function.